Validation/TrackerRecHits/src/SiStripRecHitsValid.cc

```diff
@@ -420,14 +408,17 @@ void SiStripRecHitsValid::rechitanalysis(SiStripRecHit2D const rechit,const Stri
   double mindist = 999999;
   double dist = 999999;
   PSimHit closest;
+ 
+  rechitpro.resx = -999999.; rechitpro.resy = -999999.;rechitpro.pullMF = -999999.; 
+  rechitpro.chi2 = -999999.; rechitpro.bunch = -999999.; rechitpro.event = -999999.;
   
   if(!matched.empty()){
 
-    for(vector<PSimHit>::const_iterator m=matched.begin(); m<matched.end(); m++){
-      dist = fabs(rechitpro.x - (*m).localPosition().x());
+    for(auto m : matched){
```

Comment:
There are small differences in calculation of the distance and error for the rechitsanalysis and rechitsanalysis_matched